robottelo/helpers.py

```diff
@@ -296,9 +296,11 @@ def md5_by_url(url, hostname=None):
     return result.stdout[0]
 
 
-def add_remote_execution_ssh_key(hostname, key_path=None, **kwargs):
+def add_remote_execution_ssh_key(hostname, key_path=None,
```

Comment:
This helper is also used by UI/CLI robottelo testcases. Can you make sure those are working after this change? Expecting the output for those.

Reply:
@ntkathole It should not impact anything as it works for external capsule if value provided with parameter "proxy_hostname" .
If value is "None" then default goes with Satellite only.

Here a glance if it:
+++++++++++++++++++++++++++++++++++++++++++++++
>>> External capsule way:

(Pdb) add_remote_execution_ssh_key(hostname=client.ip_addr,proxy_hostname=self.proxy_name)
2019-01-21 14:44:29 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7fc1d0f15710
2019-01-21 14:44:29 - robottelo.ssh - INFO - Connected to [capsule.example.com]
2019-01-21 14:44:29 - robottelo.ssh - INFO - >>> cat ~foreman-proxy/.ssh/id_rsa_foreman_proxy.pub
2019-01-21 14:44:31 - robottelo.ssh - INFO - <<< stdout
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC91ywLtxM84a+r8PRNwucrkTd4G/wDh5KuINABhMbz9pzQZ+NpWpb275/s9HiUL3P8ITKWehLqE40ISeUVyJLr50WYbYh8j6W0j+zzkhM5eYPiJeZS9C5Fq+jr/e8DtXxfwOW+j4eiMQttIr2jGyRTej5i1ppG1l19ZLm8KAiFzCeMLwfuGd0VUZMgI51rcmkAKa8f9q4noRoJkYH64Rwu0RNDPe1WHFQCmAPk+tQzTEAcpl2+6pcKJP+XnIo7WyjibOMtcE531O1oeBTyJgQg0kiS1PdHCY4yT1pD5xZPiJ1Te51+k4SRMpETLV/C8jTYBDFCQvWIRhhuT2Z+5fNH foreman-proxy@capsule.example.com

2019-01-21 14:44:31 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7fc1d0f15710
2019-01-21 14:44:34 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7fc1d0f21b70
2019-01-21 14:44:34 - robottelo.ssh - INFO - Connected to [192.168.123.10]
2019-01-21 14:44:34 - robottelo.ssh - INFO - >>> mkdir -p ~/.ssh
2019-01-21 14:44:36 - robottelo.ssh - INFO - >>> grep -q 'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC91ywLtxM84a+r8PRNwucrkTd4G/wDh5KuINABhMbz9pzQZ+NpWpb275/s9HiUL3P8ITKWehLqE40ISeUVyJLr50WYbYh8j6W0j+zzkhM5eYPiJeZS9C5Fq+jr/e8DtXxfwOW+j4eiMQttIr2jGyRTej5i1ppG1l19ZLm8KAiFzCeMLwfuGd0VUZMgI51rcmkAKa8f9q4noRoJkYH64Rwu0RNDPe1WHFQCmAPk+tQzTEAcpl2+6pcKJP+XnIo7WyjibOMtcE531O1oeBTyJgQg0kiS1PdHCY4yT1pD5xZPiJ1Te51+k4SRMpETLV/C8jTYBDFCQvWIRhhuT2Z+5fNH foreman-proxy@capsule.example.com' ~/.ssh/authorized_keys || echo 'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC91ywLtxM84a+r8PRNwucrkTd4G/wDh5KuINABhMbz9pzQZ+NpWpb275/s9HiUL3P8ITKWehLqE40ISeUVyJLr50WYbYh8j6W0j+zzkhM5eYPiJeZS9C5Fq+jr/e8DtXxfwOW+j4eiMQttIr2jGyRTej5i1ppG1l19ZLm8KAiFzCeMLwfuGd0VUZMgI51rcmkAKa8f9q4noRoJkYH64Rwu0RNDPe1WHFQCmAPk+tQzTEAcpl2+6pcKJP+XnIo7WyjibOMtcE531O1oeBTyJgQg0kiS1PdHCY4yT1pD5xZPiJ1Te51+k4SRMpETLV/C8jTYBDFCQvWIRhhuT2Z+5fNH foreman-proxy@capsule.example.com' >> ~/.ssh/authorized_keys
2019-01-21 14:44:37 - robottelo.ssh - INFO - >>> chmod 700 ~/.ssh
2019-01-21 14:44:39 - robottelo.ssh - INFO - >>> chmod 600 ~/.ssh/authorized_keys
2019-01-21 14:44:41 - robottelo.ssh - INFO - >>> chown -R root ~/.ssh
2019-01-21 14:44:42 - robottelo.ssh - INFO - >>> command -v restorecon && restorecon -RvF ~/.ssh || true
2019-01-21 14:44:44 - robottelo.ssh - INFO - <<< stdout
/usr/sbin/restorecon

2019-01-21 14:44:44 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7fc1d0f21b70

>>> Satellite way:

(Pdb) add_remote_execution_ssh_key(hostname=client.ip_addr)
2019-01-21 14:44:55 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7fc1d0f21b70
2019-01-21 14:44:55 - robottelo.ssh - INFO - Connected to [satellite.example.com]
2019-01-21 14:44:55 - robottelo.ssh - INFO - >>> cat ~foreman-proxy/.ssh/id_rsa_foreman_proxy.pub
2019-01-21 14:44:57 - robottelo.ssh - INFO - <<< stdout
ssh-rsa B3NzaC1yc2EAAAADAQABAAABAQC1V4poxbgYG4xRyGK0T6kGPxEVdBY/kdC5smGxuc61HqrAfhgbQ0hxk6+cES1bqTKHH5RJGeoenRAiuCNVmazAjB98jojRjnLvAAdhtewhZFvs9QPyD89vRhCXAFp0A29pBcSkycPwq4K9rltswRv3VoKaSsf0XXqcxjrfwLhqN9G3NZ+ko6WZpnzrYoTNRJss2dW09Jgj+enyFihv70EAZkKTGLINhaRPlnP1GJohWdZrYcgUre7JIoy5e82e769BEKBTLVDPcWPY4J2QV3jihJ5gr6FL2L7UgmiFc368Jf3nmN8eXeiXtg8WpPmKUox+iKr3kSQEMHO73J foreman-proxy@satellite.example.com

2019-01-21 14:44:57 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7fc1d0f21b70
2019-01-21 14:45:00 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7fc1d0f1f358
2019-01-21 14:45:00 - robottelo.ssh - INFO - Connected to [192.168.123.10]
2019-01-21 14:45:00 - robottelo.ssh - INFO - >>> mkdir -p ~/.ssh
2019-01-21 14:45:01 - robottelo.ssh - INFO - >>> grep -q 'ssh-rsa 3NzaC1yc2EAAAADAQABAAABAQC1V4poxbgYG4xRyGK0T6kGPxEVdBY/kdC5smGxuc61HqrAfhgbQ0hxk6+cES1bqTKHH5RJGeoenRAiuCNVmazAjB98jojRjnLvAAdhtewhZFvs9QPyD89vRhCXAFp0A29pBcSkycPwq4K9rltswRv3VoKaSsf0XXqcxjrfwLhqN9G3NZ+ko6WZpnzrYoTNRJss2dW09Jgj+enyFihv70EAZkKTGLINhaRPlnP1GJohWdZrYcgUre7JIoy5e82e769BEKBTLVDPcWPY4J2QV3jihJ5gr6FL2L7UgmiFc368Jf3nmN8eXeiXtg8WpPmKUox+iKr3kSQE+1x6+6MHO73J foreman-proxy@satellite.example.com' ~/.ssh/authorized_keys || echo 'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC1V4poxbgYG4xRyGK0T6kGPxEVdBY/kdC5smGxuc61HqrAfhgbQ0hxk6+cES1bqTKHH5RJGeoenRAiuCNVmazAjB98jojRjnLvAAdhtewhZFvs9QPyD89vRhCXAFp0A29pBcSkycPwq4K9rltswRv3VoKaSsf0XXqcxjrfwLhqN9G3NZ+ko6WZpnzrYoTNRJss2dW09Jgj+enyFihv70EAZkKTGLINhaRPlnP1GJohWdZrYcgUre7JIoy5e82e769BEKBTLVDPcWPY4J2QV3jihJ5gr6FL2L7UgmiFc368Jf3nmN8eXeiXtg8WpPmKUox+iKr3kSQE+1x6+6MHO73J foreman-proxy@satellite.example.com' >> ~/.ssh/authorized_keys
2019-01-21 14:45:03 - robottelo.ssh - INFO - >>> chmod 700 ~/.ssh
2019-01-21 14:45:04 - robottelo.ssh - INFO - >>> chmod 600 ~/.ssh/authorized_keys
2019-01-21 14:45:06 - robottelo.ssh - INFO - >>> chown -R root ~/.ssh
2019-01-21 14:45:07 - robottelo.ssh - INFO - >>> command -v restorecon && restorecon -RvF ~/.ssh || true
2019-01-21 14:45:09 - robottelo.ssh - INFO - <<< stdout
/usr/sbin/restorecon

2019-01-21 14:45:09 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7fc1d0f1f358
(Pdb) 

+++++++++++++++++++++++++++++++++++++++++++++++